src/Nncase.Core/IR/Expr.cs

```diff
@@ -12,11 +12,36 @@
 
 namespace Nncase.IR;
 
+/// <summary>
+/// Expression's metadata.
+/// </summary>
+public class IRMetadata
+{
+    private IReadOnlyList<string>? _outputsNames;
```

Comment:
Better to use property and naming `OutputNames`:
`public IReadOnlyList<string> OutputNames { get; set; }`

Naming `OutputNames` according to this property is a collection of `OutputName`.